src/audio/dai-zephyr.c

```diff
@@ -1215,6 +1215,8 @@ static int dai_copy(struct comp_dev *dev)
 		return 0;
 	}
 
+	dai_trigger(dd->dai->drv, dev->direction, DAI_TRIGGER_PRE_COPY);
```

Comment:
This will trigger() in every copy() ? We also should check the result.

Reply:
dmic task used to be disabled once it had finished its job
Calling it every copy now its nothing, but I'm wandering if it would become costly when MMU is introduced - drivers may work on priv level

Reply:
Dmic task used to be disabled once it has finished its job, I'm wandering if it can't become costly once MMU is introduced, drivers may work on priviledged level

Reply:
For DMIC its only one simple if and return once main job is done
if (dmic->gain == DMIC_HW_FIR_GAIN_MAX << 11)
		return;
Not a big deal, at least for now.

And I thought it is already agreed approach, thats why it was merged to upstream?

Reply:
ok, we still need to check return value and I would also except a inline comment explaining what we are doing etc,